AutoRest/Generators/Ruby/Azure.Ruby.Tests/RspecTests/lro_spec.rb

```diff
@@ -8,221 +8,433 @@
+    expect(result.body.provisioning_state).to eq("Succeeded")
+  end
+
+  it 'should succeed for delete no header in retry' do
+    result = @lros_client.delete_no_header_in_retry().value!
     expect(result.response.status).to eq(204)
   end
 
-  it 'should retry on 500 server responce in DELETE request' do
-    result = @client.lroretrys.delete_async_relative_retry_succeeded().value!
+  it 'should succeed for delete async no header in retry' do
+    result = @lros_client.delete_async_no_header_in_retry().value!
     expect(result.response.status).to eq(200)
   end
 
-  it 'should serve async POST operation' do
-    result = @client.lroretrys.post_async_relative_retry_succeeded(@product).value!
+  it 'should succeed for put sub resource' do
+    result = @lros_client.put_sub_resource(@product).value!
     expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Succeeded")
   end
 
-  it 'should return payload on POST async request' do
-    result = @client.lros.post200with_payload().value!
-    expect(result.body.id).to eq(1)
+  it 'should succeed for put async sub resource' do
+    result = @lros_client.put_async_sub_resource(@product).value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Succeeded")
+  end
+
+  it 'should succeed for put non resource' do
+    result = @lros_client.put_non_resource(@sku).value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.id).to eq('100')
+    expect(result.body.name).to eq('sku')
+  end
+
+  it 'should succeed for put async non resource' do
+    result = @lros_client.put_async_non_resource(@sku).value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.id).to eq('100')
+    expect(result.body.name).to eq('sku')
+  end
+
+  it 'should succeed for delete provisioning 202 accepted 200' do
+    result = @lros_client.delete_provisioning202accepted200succeeded().value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Succeeded")
+  end
+
+  it 'should succeed for delete provisioning 202 deleting failed 200' do
+    result = @lros_client.delete_provisioning202deleting_failed200().value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Failed")
+  end
+
+  it 'should succeed for delete provisioning 202 deleting canceled 200' do
+    result = @lros_client.delete_provisioning202deletingcanceled200().value!
+    expect(result.response.status).to eq(200)
+    expect(result.body.provisioning_state).to eq("Canceled")
+  end
+
+  it 'should succeed for delete 204' do
+    result = @lros_client.delete204succeeded().value!
+    expect(result.response.status).to eq(204)
+  end
+
+  it 'should succeed for delete 200 retry 200' do
+    result = @lros_client.delete202retry200().value!
+    expect(result.response.status).to eq(200)
+  end
+
+  it 'should succeed for delete 200 no retry 204' do

```

Comment:
corrected

---
In reply to: [69023565](https://github.com/Azure/autorest/pull/1188#discussion_r69023565) [](ancestors = 69023565)